cppwinrt/component_writers.h

```diff
@@ -468,6 +468,29 @@ catch (...) { return winrt::to_hresult(); }
             }
             else if (factory.statics)
             {
+                {
+                    auto format = R"(    template <int = 0>
+    static auto Get%Statics()
+    {
+        if constexpr (winrt::impl::has_static_lifetime_v<@::factory_implementation::%>)
```

Comment:
Am I reading this right, and the logic here is:

1. If the factory has static_lifetime, use it in the normal way
2. Otherwise, generate a null pointer and return it and call a method on that ?

Feels like this should be the other way, or some other marker beyond static-lifetime. Won't this break existing code that has static_lifetime on the factory? Maybe the check should be on the implementation type.

Similarly, where should an implementation marked static_lifetime store its state? The factory static_lifetime does instantiate the factory, it just stores it statically for the lifetime of the program.

Reply:
> Feels like this should be the other way,
It is the opposite. Currently, it always directly calls into the implementation type's static method.

> Won't this break existing code that has static_lifetime on the factory?
No. By default the CRTP base class for the factory writes non-static member functions that call into the implementation type's static method. If existing code has static_lifetime for the factory but does not have the methods directly on it, it is still a non-issue because the base class would provide those.

```cpp
struct Class : implementation::ClassT<Class>
{
    static void StaticMethod();
};

struct Class : ClassT<Class, implementation::Class, static_lifetime>
{
   // Part of the base class.
   void StaticMethod() { implementation::Class::StaticMethod(); }
};

> Similarly, where should an implementation marked static_lifetime store its state?
I don't understand your question. The state should be stored in the factory as before. Nothing has changed here.